mbed-hal/spi_api.h

```diff
@@ -64,12 +107,16 @@ extern "C" {
 /** Initialize the SPI peripheral
  *
  * Configures the pins used by SPI, sets a default format and frequency, and enables the peripheral
+ * @note The function asserts that the selected pins map onto one SPI peripheral!
+ *
  * @param[out] obj  The SPI object to initialize
  * @param[in]  mosi The pin to use for MOSI
  * @param[in]  miso The pin to use for MISO
  * @param[in]  sclk The pin to use for SCLK
+ *
+ * @return  rc  `SPI_RC_OK`
  */
-void spi_init(spi_t *obj, PinName mosi, PinName miso, PinName sclk);
+spi_return_code_t spi_init(spi_t *obj, PinName mosi, PinName miso, PinName sclk);
 
```

Comment:
So the `spi_init` function only returns `SPI_RC_OK` or get caught in `MBED_ASSERT`, so this could have also been `void`, but I wanted this to be future proof, and I don't know all the platform implementation.